accord-core/src/test/java/accord/impl/list/ListRequest.java

```diff
@@ -162,7 +209,11 @@ public ListRequest(Txn txn)
     @Override
     public void process(Node node, Id client, ReplyContext replyContext)
     {
-        node.coordinate(txn).addCallback(new ResultCallback(node, client, replyContext, txn));
+        if (id != null)
+            throw new IllegalStateException("Called process multiple times");
+        id = node.nextTxnId(txn.kind(), txn.keys().domain());
+        ((NodeSink) node.messageSink()).debugClient(id, txn, NodeSink.ClientAction.SUBMIT);
```

Comment:
Can we e.g. have some utility method for this to hide the explicit casts?

Reply:
we construct this within the `generate` method, so we could always pass in something, sure

Reply:
created a new `MessageListener` and have everything use that